### PR TITLE
Add option for enabling long press to initiate drag

### PR DESCRIPTION
**Pull request by yincrash**
in addition, i converted all the tabs to 2 spaces and added an eclipse project wrapper so it can be used as a library for eclipse built apps as well.

if you don't want the eclipse junk, I make a new pull request w/o it.

### CODE REVIEW

File: src/com/mobeta/android/dslv/DragSortListView.java [hunk 1]
```diff
@@ -466,175 +462,211 @@ private int getDragEdge(int vposition, int vtop) {
...
+    //Log.d("mobeta", "float pos="+visItemPos+" exp="+mExpDragPos+" last="+getLastVisiblePosition());
+    return visItemPos;
+
+  }
+  
+
+  @Override
+  public boolean onInterceptTouchEvent(MotionEvent ev) {
+    if (mGestureDetector == null) {
+      if (mRemoveMode == FLING || mLongPressEnabled) {
```
Comment (yincrash) [inline src/com/mobeta/android/dslv/DragSortListView.java:584]: So this is the only new code (non conversion of spaces to tabs).

Basically just use one gesture detector for both fling remove and longpress detection.

**Comment by bauerca** [comment]
Unfortunately I have to hold off on this request. There is much unexpected behavior when I use your update with the demos. For example, if a user has an OnLongClickListener attached to a list item, their listener will be called even if `dslv:long_press_enabled="true"`. This can be fixed by having long press detection in `onInterceptTouchEvent()` in DSLV. But, more importantly, item highlighting (during the long press) is messing with the floating View and the blank spot left behind while dragging. Specifically, on my 2.3 droid incredible, a ghost highlighted rectangle is left behind and the floating View is blank.

I have some ideas on how to implement long press drag initiation. Unfortunately I am preparing for a move across the country and do not have much time to work on this at the moment.

As for the Eclipse stuff, I'd rather do without it. Esp since it appears to be all formatting. Also, I would have switched to tabs instead of all double spaces :) since it seems easier to modify the appearance of tabs across different editors.

I hope I'm not sounding like a jerk; I really appreciate your input, I just want to think really carefully about this. It was quite difficult to get DSLV running as smoothly as it is now.

**Comment by kendrelaxman** [comment]
Hello  bauerca,
Thank you for the wonderful example,
Could you help me our how to add sections (Saperators) in between the List Itmes
Sorry that I am commenting here.
I appreciate your help.
Regards,
Laxmikant

**Comment by yincrash** [comment]
Sorry, I've been on vacation. (still am!)

You definitely don't sound like a jerk! This is your project to do as you wish. 

The eclipse .settings/ folder are all optional settings. I included them because they would enforce spaces instead of tabs for new committers. The only thing that is necessary is the .project file which I somehow miraculously forgot to add to my repo. Without it, other users can't link their projects to yours without either manually creating a jar with the R.\* classes (because the ant target doesn't do that) and copying over the xml in to their res/ folder or just manually copying all files into their project or creating their own eclipse wrapper again.

Alternatively, maybe you can make an ant task to handle jar-ing the R.\* classes, but users would still have to copy over the attrs.xml, I believe.

I missed the highlighting issue, because the list I use doesn't have a highlight color set.

**Comment by kendrelaxman** [comment]
Any help on my problem would be appreciated.

Thank you.

**Comment by bauerca** [comment]
@kendrelaxman You can use DSLV like a ListView for most things. In this case, I think you want the [setDivider()](http://developer.android.com/reference/android/widget/ListView.html#setDivider%28android.graphics.drawable.Drawable%29) method?

**Comment by kendrelaxman** [comment]
@bauerca,
No setDivider method will set the divider between each and every Item, while in my case I need saparator like:-

A ------> Saparator which should not be draggable
Afaganistan -----> List Item should be draggable
America-----> List Item should be draggable
B------> Saparator which should not be draggable
Bangladesh-----> List Item should be draggable
Bhutan-----> List Item should be draggable

Something like this, 
I have decided to work using DSLV but how would I achieve the above functionality in DSLV?

Thanks,
Laxmikant

**Comment by bauerca** [comment]
@kendrelaxman Aha. Now I understand. You could use list items as separators and remove the default dividers. The separator items would not be draggable if they do not contain any children with a `drag` id.

**Comment by kendrelaxman** [comment]
@bauerca,
Thanks I have managed to add the saparator in between Items which is not draggable although the ListItmes when I am dragging getting crashed, The error I am getting is Arrayout of bound exception.
If you can provide me your email Id I can send you the code.
Thanks,
Laxmikant

**Comment by bauerca** [comment]
@kendrelaxman if the crash is indeed occurring within dslv code, please open a new issue and post the traceback there along with any helpful code snippets.  thanks!

**Comment by kendrelaxman** [comment]
Hello bauerca,
I have a case where I dont want one of the Item from list should not get drag(or move from the position) having the drag id assigned to it.
Is it possible ?

**Comment by bauerca** [comment]
Done! This and more is now available in the DragSortController convenience class under the new API. Thanks!

**Comment by imminent** [comment]
That's awesome.

**Pull request closed without merging**